ck.sh

```diff
@@ -61,6 +61,9 @@ case "$COMMAND" in
 				# Copy custom skin to ckeditor-dev
 				cp -r skins/moono-lexicon ckeditor-dev/skins/moono-lexicon
 
+				# Copy lang files from image plugin to imagespacingbox
+				cp -r ckeditor-dev/plugins/image/lang plugins/imagespacingbox/
```

Comment:
Do we want to hardcode this in the build process?

Reply:
By following Julien's comment here https://github.com/liferay/liferay-ckeditor/pull/126#discussion_r480021566 , yes

Reply:
Hey @mateomustapic that comment was a suggestion, if it can be improved than lets improve it.

For the moment there's nothing "generic" that copies stuff (like directories in this case) from existing plugins to our custom plugins directories - but it would just be a matter of adding a function.

That said we don't always want to copy `lang` directories from a specific plugin to a plugin's directory.

As for 

> Do we want to hardcode this in the build process?

We're already hard coding the same kind of stuff on line 62 - and much more in that script.

Reply:
> We're already hard coding the same kind of stuff on line 62 - and much more in that script.

Yep, I know (and I don't like it xD). Everything hardcoded is related to `moono-lexicon` skin 'cause we don't expect to have more skins (but I still don't like it). I'll take a look to see if we can improve that, I think we're adding too much specific complexity to the build process.

For this one since there's no other plugin languages that need to be copied we can leave it as it is, just wanted to rise up my concern about starting to hardcore everything in the build script ;)

Reply:
> my concern about starting to hardcore everything

![hardcore](https://user-images.githubusercontent.com/7074/91843072-62ab7300-ec55-11ea-92c2-365954f44f54.gif)

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2F4WEVDuEbYosIbjnTBG%2Fgiphy.gif&token=hardcore" data-gitmeme-token="hardcore"><img src="https://media1.giphy.com/media/4WEVDuEbYosIbjnTBG/giphy.gif" title="Created by gitme.me with /hardcore"/></a>

Reply:
@carloslancha yes it's not perfect, but it works, I agree that in the long term if we write more and more plugins we'll need something that handles this "automagically"

We can think of what should be done for each plugin (it might seem like an overkill) - that would probably mean writing something like we did for the plugin's dependencies in the `codemirror` plugin, maybe a `plugin.json` file (instead of our good old `deps.json`)

```json
{
  "name": "someplugin",
  "dependencies": [
      "foo": "path/to/foo",
      "bar": "path/o/bar"
   ],
   "languages": [
        "path/to/ckeditor-dev/plugins/someplugin/lang",

    ],
}
```

The we can have a JS script (because you don't like bash right :joy: ) parse the json and do it's magic.

But I think we should be careful about this

Reply:
A great philosopher once said:

> Dumb is the new clever.

Or maybe it was:

> Clever is the new dumb.

Actually, now that I think about it, it wasn't a great philosopher; it was just some guy I used to work with and ever since I heard it I have been repeating it whenever I get the chance. 😂

Playing on Knuth's famous "Premature optimization is the root of all evil", ~another great philosopher~ some person on Hacker News once quipped:

> Premature Abstraction is the root of all evil.

To which [another commenter replied](https://news.ycombinator.com/item?id=9442140):

> Premature optimization, that's like a fart. Premature abstraction is like taking a dump on another developer's desk.

All of which is to say, beware of over-engineering this.

Reply:
I agree, that's why I said that it might seem like an overkill ...

Reply:
Quoting myself 🤷 :
> For this one since there's no other plugin languages that need to be copied we can leave it as it is

Reply:
![friends](https://user-images.githubusercontent.com/7074/91955369-5d0c6680-ed03-11ea-8a4e-162db2ad3e80.gif)

Reply:
In conclusion, as I understand, this plugin should be done now and doesn't require any more work?

Reply:
Could you please rebase with master and push? Don't know why is not getting last merged changed to check the format on plugins

Reply:
Rebased and pushed, please check now.

Reply:
I think this is not being build with the `--no-ie-checks` but we can take a look on master after merging this one. LGTM

Reply:
Hey @mateomustapic I'm rebasing this pr since have conflicts with recent master due to the changes introduced by https://github.com/liferay/liferay-ckeditor/pull/128

While doing it I realize that we are copying the language files directly in our working plugins folder, that will result in files that we will need to commit or discard, but I think what we want to do is copy them to the building folder (`ckeditor-dev` here, `temp` after rebasing, so

```
cp -r ckeditor-dev/plugins/image/lang plugins/imagespacingbox/
```
will become
```
cd ckeditor-dev
cp -r plugins/images/lang plugins/imagespacingbox
```
and after rebase
```
cd temp
cp -r plugins/images/lang plugins/imagespacingbox
```
Does this make sense to  you?

Reply:
Hi, yes I see this generates language files to both `ckeditor-dev/plugins` and working `plugins` directory.
Will you make this change to the script or should I?

Reply:
I did it already while rebasing with master and pushed to this branch. Please, test the pr with the latest changes and if everything is working as expected, since @julien approved it, feel free to merge it. Thx.